File: VB/WebSampleVB/WebSample.Module.Web/WebTreeViewActionNode.vb
Imports Microsoft.VisualBasic
Imports DevExpress.ExpressApp.Actions
Imports System.Web.UI.WebControls
Imports DevExpress.ExpressApp.Utils

Namespace WebSample.Module.Web
	Public Class WebTreeViewActionNode
		Inherits TreeNode
		Private action As SingleChoiceAction
		Private itemValue As ChoiceActionItem
		Public Sub New(ByVal action As SingleChoiceAction, ByVal itemValue As ChoiceActionItem, ByVal owner As TreeView)
			MyBase.New(owner, False)
			Me.action = action
			Me.itemValue = itemValue
			If itemValue IsNot Nothing Then
				Me.Text = itemValue.Caption
				Me.ImageUrl = ImageLoader.Instance.GetImageInfo(itemValue.Info.GetAttributeValue(DevExpress.ExpressApp.NodeWrappers.VisualItemInfoNodeWrapper.ImageNameAttribute)).ImageUrl
				Me.SelectAction = TreeNodeSelectAction.Select
			End If
		End Sub
		Public Sub ExecuteAction()
			action.DoExecute(itemValue)
		End Sub
	End Class
End Namespace

File: CS/WebSample.Module.Web/WebTreeViewActionNode.cs
using DevExpress.ExpressApp.Actions;
using System.Web.UI.WebControls;
using DevExpress.ExpressApp.Utils;

namespace WebSample.Module.Web {
    public class WebTreeViewActionNode : TreeNode {
        private SingleChoiceAction action;
        private ChoiceActionItem itemValue;
        public WebTreeViewActionNode(SingleChoiceAction action, ChoiceActionItem itemValue, TreeView owner)
            : base(owner, false) {
            this.action = action;
            this.itemValue = itemValue;
            if (itemValue != null) {
                this.Text = itemValue.Caption;
                this.ImageUrl = ImageLoader.Instance.GetImageInfo(itemValue.Info.GetAttributeValue(DevExpress.ExpressApp.NodeWrappers.VisualItemInfoNodeWrapper.ImageNameAttribute)).ImageUrl;
                this.SelectAction = TreeNodeSelectAction.Select;
            }
        }
        public void ExecuteAction() {
            action.DoExecute(itemValue);
        }
    }
}
